nextest-filtering/src/lib.rs

```diff
@@ -0,0 +1,271 @@
+// Copyright (c) The nextest Contributors
+// SPDX-License-Identifier: MIT OR Apache-2.0
+
+use std::{cell::RefCell, collections::HashSet};
+
+use guppy::{
+    graph::{DependsCache, PackageGraph, PackageMetadata},
+    PackageId,
+};
+
+pub mod error;
+mod parsing;
+
+use parsing::{Expr, RawNameMatcher, SetDef};
+
+/// Matcher for name
+///
+/// Used both for package name and test name
+#[derive(Debug, Clone)]
+pub enum NameMatcher {
```

Comment:
Could you move the contents of lib.rs into another file? The reason is that anything that's private here is fully accessible to any other modules.